src/hooks/safe-apps/useSafeAppsFilters.ts

```diff
@@ -29,6 +30,14 @@ const useSafeAppsFilters = (safeAppsList: SafeAppData[]): ReturnType => {
 
   const filteredApps = useAppsFilterByOptimizedForBatch(filteredAppsByQueryAndCategories, optimizedWithBatchFilter)
 
+  const debouncedSearchQuery = useDebounce(query, 500)
```

Comment:
Just a thought: what about tracking this on unmount instead, e.g. when the chosen app is open? This is highly dependent on the speed of the typing.

Reply:
We might miss some data points then i.e. users who search for an app but don't open it. Also I think we already track whenever a safe app is open via the `SAFE_APPS_EVENTS.OPEN_APP` event.

Reply:
Maybe 2 seconds then?

Reply:
Or better, store the list of distinct queries, and track on unmount.
If a query starts with the same symbols as an existing query, overwrite it instead of adding to the set.

If we end up with stuff like `gov`, `govern`, `safe gover` all tracked in the same session, that would be too much.

Reply:
A bigger debounce should be OK too, though. Maybe 5 seconds.
The search is also pretty slow and blocks typing.

Reply:
> We might miss some data points then i.e. users who search for an app but don't open it.

The only time that would happen is if the user closes the browser, otherwise unmount always occurs when the user navigates within the app. It's certainly something to consider though.

> Also I think we already track whenever a safe app is opened via the `SAFE_APPS_EVENTS.OPEN_APP` event.

I'm thinking about over-tracking of the search term. Perhaps `onBlur` would be a better reference point as the `useEffect` may run >1 times for slow typers. What do you think?

Reply:
I noticed that there already was a debounce of 400ms in place so I removed my change and adjusted the other hook instead. So currently, the safe apps list is updated after 400ms of not typing and thats where we also dispatch the analytics event. We could increase that number but imo it would make the UX worse because it would take longer to see the results of the search.

Was this call removed because of over-tracking? If not I suggest we restore the previous state and create a separate ticket for future improvements on the mechanism.

Reply:
I can imagine that if the list updates too fast it could "interrupt" the user. I assume that it could also lead to users not typing out longer search terms anymore because they already see a result. There is [some interesting info](https://ux.stackexchange.com/questions/95336/how-long-should-the-debounce-timeout-be) on this topic which I think we should explore before making such a change.

As for the analytics, lets add a second debounce (2s) as suggested.

Reply:
I think there's some inefficiency in the re-rendering too, it shouldn't be so slow. In Safari, the search bar is super sluggish.